pkg/defaults/defaults.go

```diff
@@ -98,7 +98,7 @@ const (
 	DefaultMapRoot = "/sys/fs/bpf"
 
 	// DefaultCgroupRoot is the default path where cilium cgroup2 should be mounted
-	DefaultCgroupRoot = "/var/run/cilium/cgroupv2"
+	DefaultCgroupRoot = "/run/cilium/cgroupv2"
```

Comment:
Any time that we change default options, we have the potential to impact upgrade, and the consequences are not always immediately clear. Not sure if there was any discussion on this particular change, but I've thought it through a bit below.

In practice this should not have any impact, unless there's the unusual combination of:
* Cilium is _not_ run in a container.
* The underlying distribution does _not_ link `/var/run` to `/run`.

Maybe in the above case, upgrading Cilium to this version with a new default could leave an extra mount of the cgroupfs behind which would never be unmounted. The user could of course unmount it.

This is all to say basically, I think the change is probably fine, but in future I'd suggest avoiding changes to defaults like this unless there's a clear reason for it and it's accompanied with upgrade impact analysis.

Reply:
@joestringer FWIW, I did consider upgrade/downgrade implications before making the change. 

>upgrading Cilium to this version with a new default could leave an extra mount of the cgroupfs behind which would never be unmounted.

How would that happen? 🤔  `/var/run` is a symlink to `/run` on most platforms.

Reply:
It's good to know that you had considered upgrade/downgrade implications. It can be a bit hard to gauge whether different developers are considering this on their PRs and whether we all have a shared understanding of the impacts. As someone trying to assess whether we should be including particular PRs in upcoming releases, my goal is basically to minimize the risk of the changes. If I see something that appears risky through some fuzzy logic (glancing at the changes, looking at what CI is saying) then I tend to look into what I can do to reduce that risk. The above comment was the output of that process with basically an assessment (low risk) and a suggestion (maybe in future we can avoid making changes to pkg/defaults).

One of the reasons I raised the comment above is that I see any change to `pkg/defaults` in a backport patch as a red flag. In general our upgrade model tends to assume that agent defaults do not change, with the notion in mind that users can end up using their existing DS YAMLs and just doing an image tag bump and they will not adopt new changes to existing settings. On the contrary if we want _new_ users to adopt some setting, then we tend to set the new value in the helm configuration and avoid making the change in the daemon. I'm not sure if we've ever formally encoded this knowledge somewhere or whether it's been effectively passed around the core team. (If we have passed this on successfully, then bear with me explaining it again :sweat_smile: )

The reason for this approach is that there's a certain class of upgrade bugs that can be caused by changing agent defaults. On one hand, banning changes to `pkg/defaults` in bugfix PRs isn't going to get rid of upgrade breakages. On the other hand, if it's possible to avoid changing default settings then I think we should avoid making those changes, so we don't have to consider this class of upgrade bugs.